### PR TITLE
feat: add validation for long path support on windows

### LINKED ISSUE #81: Add validation for long path support
Validate for the presence of the Windows registry key which [enables long path support](https://github.com/unoplatform/uno/discussions/9370).

The suggestion should also include a mention to reboot once the change has been made.

### PR DESCRIPTION
**Pull request by aesteves900**
GitHub Issue (If applicable): closes #81 

# PR Type

What kind of change does this PR introduce?

- Feature

## What is the current behavior?

People from the community have reported that certain projects fail to build because of long paths.

This is a longstanding issue affecting Win32.

## What is the new behavior?

The uno-check will apply the registry changes.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] Tested code with current Figma desktop app
- [ ] Docs have been added/updated which fit [documentation template](https://github.com/nventive/Uno/blob/master/doc/.feature-template.md). (for bug fixes / features)
- [x] Unit Tests for the changes have been added (for bug fixes / features) (if applicable)
- [x] [Wasm UI Tests](doc/articles/working-with-the-samples-apps.md#running-the-webassembly-ui-tests-snapshots) are not showing unexpected any differences. Validate PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes and the __PlugIn is BACKWARD COMPATIBLE__
- [x] Associated with an issue

### CODE REVIEW

File: UnoCheck/Solutions/LongPathsNotEnabledSolution.cs [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+			_dword = dword;
+		}
+
+		public override async Task Implement(SharedState sharedState, CancellationToken cancellationToken)
+		{
+			await base.Implement(sharedState, cancellationToken);
+
+			_fileSystemKey.SetValue(_dword, 1);
+
+			ReportStatus("The long paths are enabled");
```
Comment (jeromelaban) [inline UnoCheck/Solutions/LongPathsNotEnabledSolution.cs:22]:
```suggestion
			ReportStatus("The long paths are enabled, you will need to reboot your machine for the change to take effect.");
```

File: UnoCheck/Checkups/WindowsRegistryCheckup.cs [hunk 2]
```diff
@@ -0,0 +1,40 @@
+﻿#nullable enable
+
+using DotNetCheck.Models;
+using DotNetCheck.Solutions;
+using Microsoft.Win32;
+using System.Threading.Tasks;
+
+namespace DotNetCheck.Checkups
+{
+	public class WindowsRegistryCheckup : Checkup
```
Comment (jeromelaban) [inline UnoCheck/Checkups/WindowsRegistryCheckup.cs:10]: Let's make it explicitly about checking the long path support:

```suggestion
	public class WindowsLongPathCheckup : Checkup
```

File: UnoCheck/Checkups/WindowsRegistryCheckup.cs [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+	{
+		public override string Id => "wregistry";
+
+		public override string Title => "Windows Registry Checkup";
```
Comment (jeromelaban) [inline UnoCheck/Checkups/WindowsRegistryCheckup.cs:14]:
```suggestion
		public override string Title => "Windows Long Path Checkup";
```

File: UnoCheck/Checkups/WindowsRegistryCheckup.cs [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+using DotNetCheck.Models;
+using DotNetCheck.Solutions;
+using Microsoft.Win32;
+using System.Threading.Tasks;
+
+namespace DotNetCheck.Checkups
+{
+	public class WindowsRegistryCheckup : Checkup
+	{
+		public override string Id => "wregistry";
```
Comment (jeromelaban) [inline UnoCheck/Checkups/WindowsRegistryCheckup.cs:12]:
```suggestion
		public override string Id => "windowslongpath";
```

File: UnoCheck/Solutions/LongPathsNotEnabledSolution.cs [hunk 5]
```diff
@@ -0,0 +1,28 @@
+﻿using DotNetCheck.Models;
+using Microsoft.Win32;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace DotNetCheck.Solutions
+{
+	public class LongPathsNotEnabledSolution : Solution
+	{
+		public LongPathsNotEnabledSolution(RegistryKey fileSystemKey, string dword) : base()
```
Comment (jeromelaban) [inline UnoCheck/Solutions/LongPathsNotEnabledSolution.cs:10]:
```suggestion
		public LongPathsNotEnabledSolution(RegistryKey fileSystemKey, string keyName)
```

File: UnoCheck/Checkups/WindowsRegistryCheckup.cs (hunk 2 above)
Comment (aesteves900) [reply UnoCheck/Checkups/WindowsRegistryCheckup.cs]: Done!

File: UnoCheck/Checkups/WindowsRegistryCheckup.cs (hunk 4 above)
Comment (aesteves900) [reply UnoCheck/Checkups/WindowsRegistryCheckup.cs]: Done!

**Review by jeromelaban — APPROVED** [review]
Thanks !!

**Pull request merged**